internal/pkg/config/config.go

```diff
@@ -140,7 +307,33 @@ func LoadInjectionConfig(reader io.Reader) (*InjectionConfig, error) {
 		return nil, ErrMissingName
 	}
 
-	glog.V(3).Infof("Loaded injection config %s sha256sum=%x", cfg.Name, sha256.Sum256(data))
+	// we need to split the Name field apart into a Name and Version component
+	cfg.Name, cfg.version = configNameFields(cfg.Name)
+
+	glog.V(3).Infof("Loaded injection config %s version=%s sha256sum=%x", cfg.Name, cfg.Version(), sha256.Sum256(data))
 
 	return &cfg, nil
 }
+
+// given a name of a config, extract the name and version. Format is "name[:version]" where :version
+// is optional, and is assumed to be "latest" if omitted.
+func configNameFields(shortName string) (name, version string) {
+	substrings := strings.Split(shortName, ":")
```

Comment:
`substrings := strings.SplitN(shortName, ":", 2)` will be safer here so that something like `my-injector:v3:latest` does not throw this function off. Not sure is you want to enforce some kind of format of the name in general elsewhere.